app/helpers/publishers_helper.rb

```diff
@@ -43,8 +43,10 @@ def new_publisher?(publisher)
     is_new.present? && publisher.channels.size.zero?
   end
 
-  def payout_in_progress?
-    !!Rails.cache.fetch(SetPayoutInProgressJob::PAYOUT_IN_PROGRESS)
+  def payout_in_progress?(publisher)
+    selected_wallet = publisher.selected_wallet_provider
+    payouts_in_progress = Rails.cache.fetch(SetPayoutsInProgressJob::PAYOUTS_IN_PROGRESS)
+    selected_wallet.present? && payouts_in_progress.present? && payouts_in_progress[publisher.selected_wallet_provider.class.name.underscore]
```

Comment:
It seems we need this logic in multiple places, the whole freeze/underscore/classname. Can we extract a simple service class that centralizes this logic?

Then the views could call it, i.e. `PayoutService.payout_in_progress?(creator)`
and the job could call it, i.e. `PayoutService.set_all_in_progress()`

It would look basically like `set_payouts_in_progress_job.rb` 

WARNING NOT TESTED
```ruby
class PayoutService
  PAYOUTS_IN_PROGRESS = 'payouts_in_progress'.freeze

  CONNECTIONS = [BitflyerConnection.name.underscore,
                 GeminiConnection.name.underscore,
                 PaypalConnection.name.underscore,
                 UpholdConnection.name.underscore].freeze

  def self.set_all_in_progress
    Rails.cache.write(PAYOUTS_IN_PROGRESS,
                      Hash[CONNECTIONS.collect { |connection| [connection, true] }])
  end

  def self.payout_in_progress?(creator)
    selected_wallet = publisher.selected_wallet_provider
    payouts_in_progress = Rails.cache.fetch(PAYOUTS_IN_PROGRESS)
    selected_wallet.present? && payouts_in_progress.present? && 
      payouts_in_progress[creator.selected_wallet_provider.class.name.underscore]
  end
end
```

Thoughts?

Reply:
Good idea! Will update

Reply:
I'm going to keep logic in here since the convention is to have the helper handle the logic, but I did make it DRY.